packages/icon-button/src/IconButton.tsx

```diff
@@ -234,31 +247,60 @@ const getIconStyle = (size: Size) => css`
  * @param props.variant Determines color of `IconButton`. Can be `light` or `dark`.
  * @param props.href Destination URL, if supplied `IconButton` will render in `a` tags, rather than `button` tags.
  * @param props.onClick Callback fired when `IconButton` is clicked.
- * @param props.ariaLabel Required prop that will be passed to `aria-label` attribute
  * @param props.active Determines whether `IconButton` will appear `active`
  *
  */
 
 const IconButton = React.forwardRef((props: IconButtonProps, ref) => {
   const {
-    variant = 'light',
+    variant = Variant.Light,
+    size = Size.Default,
     disabled = false,
-    size = 'default',
     active = false,
     className,
     href,
     children,
-    ariaLabel,
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledBy,
     ...rest
   } = props;
 
-  const renderIconButton = (Root: React.ElementType<any> = 'button') => (
+  // We do our own proptype validation here to ensure either 'aria-label' or 'aria-labelledby' are passed to the component.
+  if (!ariaLabel && !ariaLabelledBy) {
+    console.error(
```

Comment:
A ticket for us to do in the future is to make our own leafygreen.error function that does what React does and doesn't show these warnings in production.

Reply:
True. Right now, we're super reliant on consuming apps stripping out console messages in their build.

https://jira.mongodb.org/browse/PD-600